app/src/main/java/hr/dice/filipbionda/tmdbpractice/ui/homescreen/HomeScreen.kt

```diff
@@ -0,0 +1,567 @@
+                            modifier = Modifier.width(dimensionResource(R.dimen.search_bar_suggestion_content_padding)),
+                        )
+                        Text(
+                            text = suggestion,
+                            style = MaterialTheme.typography.bodySmall,
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun HomeScreenChipGroup(
+    categories: List<ContentType>,
+    onChipClick: (ContentType) -> Unit,
+    modifier: Modifier = Modifier,
+    initialCategory: ContentType = ContentType.MOVIE,
+) {
+    var selectedChip by rememberSaveable {
+        mutableStateOf(initialCategory)
+    }
+    LazyRow(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier =
+        modifier
+            .fillMaxWidth(),
+    ) {
+        items(categories) { category ->
+            FilterChip(
+                modifier =
+                Modifier
+                    .padding(
+                        start =
+                        if (category == initialCategory) {
+                            dimensionResource(
+                                R.dimen.home_screen_horizontal_padding,
+                            )
+                        } else {
+                            dimensionResource(R.dimen.filter_chip_group_content_padding)
+                        },
+                    )
+                    .height(30.dp)
+                    .width(90.dp),
+                selected = selectedChip == category,
+                shape = RoundedCornerShape(dimensionResource(R.dimen.filter_chip_shape_size)),
+                border = null,
+                colors =
+                FilterChipDefaults.filterChipColors(
+                    selectedContainerColor = secondaryColor,
+                    selectedLabelColor = MaterialTheme.typography.labelSmall.color,
+                    disabledLabelColor = Color.White,
+                    disabledSelectedContainerColor = Color.Transparent,
+                ),
+                onClick = {
+                    selectedChip = category
+                    onChipClick(selectedChip)
+                },
+                label = {
+                    Text(
+                        text = createChipLabel(category),
+                        style = if (category == selectedChip) {
+                            MaterialTheme.typography.labelSmall
+                        } else {
+                            MaterialTheme.typography.labelSmall.copy(
+                                color = grey_50,
+                                fontWeight = FontWeight.Normal,
+                            )
+                        },
+                        textAlign = TextAlign.Center,
+                        modifier = Modifier.fillMaxWidth(),
+                    )
+                },
+            )
+        }
+    }
+}
+
+@Composable
+fun HomeScreenContent(
+    items: List<String>,
```

Comment:
For lists use Immutable collections from https://github.com/Kotlin/kotlinx.collections.immutable

Reply:
Recomposition happens for numerous reasons in Jetpack Compose. Compose marks your parameters as `@Stable` or `@Unstable`. If you use `@Unstable` values they will always get recomposed because compose compiler cannot verify that they in fact have not changed value. Immutable collections are stable and Lists are not. You can find more about it in this [blog post](https://medium.com/androiddevelopers/new-ways-of-optimizing-stability-in-jetpack-compose-038106c283cc)